torch/onnx/utils.py

```diff
@@ -1787,64 +1758,58 @@ def _run_symbolic_function(
 
-        if namespace == "aten":
-            domain = ""
-        elif namespace == "quantized" and symbolic_helper.is_caffe2_aten_fallback():
+        if namespace == "quantized" and symbolic_helper.is_caffe2_aten_fallback():
             domain = "caffe2"
         else:
             domain = namespace
+        symbolic_function_name = f"{domain}::{op_name}"
 
-        if symbolic_registry.is_registered_op(op_name, domain, opset_version):
-            symbolic_fn = _find_symbolic_in_registry(
-                domain, op_name, opset_version, operator_export_type
-            )
-            assert symbolic_fn is not None
-
-            attrs = {k: symbolic_helper._node_get(n, k) for k in n.attributeNames()}
-            if _need_symbolic_context(symbolic_fn):
-                ctx = _exporter_states.SymbolicContext(_params_dict, env, n, block)
-                return symbolic_fn(ctx, g, *inputs, **attrs)
-            # PythonOp symbolic need access to the node to resolve the name conflict,
-            # this is inconsistent with regular op symbolic.
-            if op_name == "PythonOp":
-                inputs = (n, *inputs)
-            return symbolic_fn(g, *inputs, **attrs)
-        elif namespace == "onnx":
+        symbolic_function_group = registration.registry.get_function_group(
+            symbolic_function_name
+        )
+        if symbolic_function_group is not None:
+            symbolic_fn = symbolic_function_group.get(opset_version)
+            if symbolic_fn is not None:
+                attrs = {k: symbolic_helper._node_get(n, k) for k in n.attributeNames()}
+                if _need_symbolic_context(symbolic_fn):
+                    # TODO(justinchuby): Refactor how we check for the need of the symbolic context
+                    ctx = _exporter_states.SymbolicContext(_params_dict, env, n, block)
+                    return symbolic_fn(ctx, g, *inputs, **attrs)
+                # PythonOp symbolic need access to the node to resolve the name conflict,
+                # this is inconsistent with regular op symbolic.
+                if op_name == "PythonOp":
+                    inputs = (n, *inputs)
+                return symbolic_fn(g, *inputs, **attrs)
+
+        attrs = {
+            k + "_" + n.kindOf(k)[0]: symbolic_helper._node_get(n, k)
+            for k in n.attributeNames()
+        }
+        if namespace == "onnx":
             # Clone node to trigger ONNX shape inference
-            attrs = {
-                k + "_" + n.kindOf(k)[0]: symbolic_helper._node_get(n, k)
-                for k in n.attributeNames()
-            }
             return g.op(op_name, *inputs, **attrs, outputs=n.outputsSize())  # type: ignore[attr-defined]
-        elif _should_aten_fallback(
-            namespace, op_name, opset_version, operator_export_type
-        ):
+
+        if _should_aten_fallback(ns_op_name, opset_version, operator_export_type):
```

Comment:
:( I see why this isn't `symbolic_function_name`

Reply:
I wonder if we can use the same name. I didn’t think too deeply about it. Just preserved the old behavior

Reply:
We can do this later per discussion